modules/SofaConstraint/FreeMotionAnimationLoop.cpp

```diff
@@ -218,6 +218,10 @@ void FreeMotionAnimationLoop::step(const sofa::core::ExecParams* params, SReal d
         freePosEqPosPlusFreeVelDt.setMapped(true);
         getContext()->executeVisitor(&freePosEqPosPlusFreeVelDt);
     }
+	
+	mop.projectPositionAndVelocity(freePos, freeVel); // apply projective constraints
+	mop.propagateXAndV(freePos, freeVel);
```

Comment:
this is wrong, it will linearize mappings around free motion instead of beginning of time step position

Reply:
I am sorry that I have made a wrong fix since this issue has not been sorted out for a long time.
(｡ŏ_ŏ)

Reply:
I have tried your modification. However it does not work.